src/main/java/io/confluent/idesidecar/restapi/proxy/clusters/strategy/ConfluentLocalKafkaClusterStrategy.java

```diff
@@ -1,45 +1,64 @@
+   * associated with.
+   * @param context The context of the proxy request.
+   * @return The headers to be passed to our implementation of the Kafka REST API.
+   */
+  @Override
+  public MultiMap constructProxyHeaders(ClusterProxyContext context) {
+    return HttpHeaders
+        .headers()
+        .add(CONNECTION_ID_HEADER, context.getConnectionId());
   }
 
-
+  /**
+   * Route the request back to ourselves at the /internal/kafka path.
+   * Context: We used to send this proxy request to the Kafka REST server running in the
+   * confluent-local container, but now we route the request to our own implementation of the
+   * Kafka REST API, served at the /internal/kafka path. This was done to get early feedback
+   * on our in-house implementation of the Kafka REST API.
+   * @param requestUri The URI of the incoming request.
+   * @param clusterUri The Kafka REST URI running alongside the Kafka cluster.
+   *                   (unused here)
+   * @return The URI of the Kafka REST API running in the sidecar.
+   */
   @Override
   public String constructProxyUri(String requestUri, String clusterUri) {
-    // Remove the /kafka prefix from the request URI since this is how the REST API
-    // running in Confluent Local Kafka is configured.
-    return uriUtil.combine(clusterUri, requestUri.replaceFirst("^(/kafka|kafka)", ""));
+    return uriUtil.combine(
+        sidecarHost, requestUri.replaceFirst("^(/kafka|kafka)", "/internal/kafka")
+    );
   }
 
+  /**
+   * In addition to replacing the cluster URLs with the sidecar host, we also need to replace
+   * the internal Kafka REST path /internal/kafka with the external facing /kafka path.
+   * @param proxyResponseBody The response body from the Kafka cluster.
+   * @return The response body with the internal Kafka REST path replaced with the external path.
+   */
   @Override
-  public String processProxyResponse(String proxyResponse, String clusterUri, String sidecarHost) {
-    return super.processProxyResponse(
-        proxyResponse,
-        // We don't care for the clusterUri passed to this method since it
-        // points to the externally accessible URI of the Confluent Local Kafka REST API
-        // exposed from the running docker container.
-        // Instead, the response contains the docker container's internal hostname ("rest-proxy"),
-        // which is what we want to replace with the sidecar URI.
-        "http://%s".formatted(confluentLocalKafkaRestHostname),
-        // Since the REST API running in Confluent Local Kafka is configured to run on /v3,
-        // we add the /kafka to the sidecar URI before replacing the cluster URI
-        // with the sidecar URI.
-        uriUtil.combine(sidecarHost, "/kafka")
-    );
+  public String processProxyResponse(String proxyResponseBody) {
+    return super
+        .processProxyResponse(proxyResponseBody)
+        .replaceAll("/internal/kafka", "/kafka");
```

Comment:
This is a lot less specific than including `https://<host>`. Should we also include that, just because we're doing a blind string-based replace-all on the response?

Reply:
Do we need to perform the replacement? I mean, technically, the `/internal/kafka/...` routes are valid and working.

Reply:
I think we should perform the replacement given that we want to (at least initially) hide the `/internal/kafka` routes from ide-sidecar clients.